### ISSUE TITLE
TypeError: n is not iterable occurs when clicking span in grafana flame graph

### ISSUE THREAD

**Issue by huangyutongs** [deepflowio/deepflow#5826]
### Search before asking

- [X] I had searched in the [issues](https://github.com/deepflowio/deepflow/issues?q=is%3Aissue) and found no similar feature requirement.

### DeepFlow Component

Grafana Dashbaord

### What you expected to happen

I can click on the span in the flame graph normally

### How to reproduce

In the Distributed Tracing - Cloud panel, the vm option is Any, the request_resource option adds a filter /api/app*, select any one in the Request List, return to the flame graph and click one of the system types span, the panel will report an error.
![image](https://github.com/deepflowio/deepflow/assets/48778924/5c2dabda-2c9e-4345-8b2d-6935e5de6075)

![image](https://github.com/deepflowio/deepflow/assets/48778924/000374d4-1471-4d32-a290-6eb1eaaa838e)

### DeepFlow version

deepflow-server -v
```
2024/03/18 11:24:01 ENV K8S_NODE_NAME_FOR_DEEPFLOW=iu54-k8s-sit-worker1; K8S_NODE_IP_FOR_DEEPFLOW=192.168.120.54; K8S_POD_NAME_FOR_DEEPFLOW=deepflow-server-6bd94ccb99-m4vmk; K8S_POD_IP_FOR_DEEPFLOW=10.42.5.86; K8S_NAMESPACE_FOR_DEEPFLOW=deepflow
Name: deepflow-server community edition
Branch: v6.4
CommitID: 19c04830fabdfd87a21bc8bdf0ad59a954890340
RevCount: 9710
Compiler: go version go1.20.14 linux/amd64
CompileTime: 2024-03-07 02:39:30
```
deepflow-agent -v
```
9706-fa82985253d91d03b1cdb5617c9c4b7381996b86
Name: deepflow-agent community edition
Branch: v6.4
CommitId: fa82985253d91d03b1cdb5617c9c4b7381996b86
RevCount: 9706
Compiler: rustc 1.75.0 (82e1608df 2023-12-21)
CompileTime: 2024-03-05 08:07:54
```

### DeepFlow agent list

```
root@iu51-k8s-sit-master1:~# deepflow-ctl agent list
ID   NAME                      TYPE       CTRL_IP          CTRL_MAC            STATE    GROUP         EXCEPTIONS   REVISION    UPGRADE_REVISION   
1    iu55-k8s-sit-worker2-V3   K8S_VM     192.168.120.55   00:50:56:93:ae:33   NORMAL   default                    v6.4 9706                      
2    iu57-k8s-sit-worker4-V7   K8S_VM     192.168.120.57   00:50:56:93:7f:d7   NORMAL   default                    v6.4 9706                      
3    iu56-k8s-sit-worker3-V8   K8S_VM     192.168.120.56   00:50:56:93:fa:8b   NORMAL   default                    v6.4 9706                      
4    iu51-k8s-sit-master1-V1   K8S_VM     192.168.120.51   00:50:56:93:50:a0   NORMAL   default                    v6.4 9706                      
5    iu53-k8s-sit-master3-V2   K8S_VM     192.168.120.53   00:50:56:93:06:73   NORMAL   default                    v6.4 9706                      
6    iu58-k8s-sit-worker5-V4   K8S_VM     192.168.120.58   00:50:56:93:db:5e   NORMAL   default                    v6.4 9706                      
7    iu52-k8s-sit-master2-V5   K8S_VM     192.168.120.52   00:50:56:93:ee:f9   NORMAL   default                    v6.4 9706                      
8    iu54-k8s-sit-worker1-V6   K8S_VM     192.168.120.54   00:50:56:93:f2:1d   NORMAL   default                    v6.4 9706                      
9    IU25-YHWMS-SIT-MYSQL-W9   CHOST_VM   192.168.120.25   00:50:56:b1:1a:18   NORMAL   legacy-host                v6.4 9724
```

### Kubernetes CNI

cilium
Client: 1.14.0 b5013e15 2023-07-26T10:28:38+02:00 go version go1.20.5 linux/amd64
Daemon: 1.14.0 b5013e15 2023-07-26T10:28:38+02:00 go version go1.20.5 linux/amd64

### Operation-System/Kernel version

```
"Ubuntu 22.04.3 LTS"
5.15.0-78-generic
```

### Anything else

When reproducing, no error logs occurred in grafana, deepflow-server, and clickhouse.

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/deepflowio/deepflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by twou12031** [comment]
THX for report.

Working on this issue, but need more information.
Could you comment the data that has the same problem or send it to our community maintenance staff?
This data will only be used for testing purposes.


Here's how you can find that data:

step.1: go to the view panel page
<img width="1509" alt="image" src="https://github.com/deepflowio/deepflow/assets/30427789/dcf338c0-083f-43a0-a205-99b4f69319c3">

step.2: open the [ Inspect:Panel ] drawer
<img width="1512" alt="image" src="https://github.com/deepflowio/deepflow/assets/30427789/9faf2ec2-4601-430d-aa3d-2b589f86af9c">

step.3: download CSV
<img width="1512" alt="image" src="https://github.com/deepflowio/deepflow/assets/30427789/4c56eb1c-ffa8-4376-b488-8d6d6959608a">

**Comment by huangyutongs** [comment]
> THX for report.
> 
> Working on this issue, but need more information. Could you comment the data that has the same problem or send it to our community maintenance staff? This data will only be used for testing purposes.
> 
> Here's how you can find that data:
> 
> step.1: go to the view panel page <img alt="image" width="1509" src="https://private-user-images.githubusercontent.com/30427789/314371939-dcf338c0-083f-43a0-a205-99b4f69319c3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTEzNTUwMTIsIm5iZiI6MTcxMTM1NDcxMiwicGF0aCI6Ii8zMDQyNzc4OS8zMTQzNzE5MzktZGNmMzM4YzAtMDgzZi00M2EwLWEyMDUtOTliNGY2OTMxOWMzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMjUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzI1VDA4MTgzMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWJjODUzYzY5ZTE4MzQzOTcwZmU1ZDdmNTg1YzVjYjE2OTZiYzIzZjMwYWMzYjRjNGIyM2IwNTVhMDdiMjcwNGQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.6hcad-F7noP3CcFP3wwDRBs0wwSvIUxBmck3eLJ2YBE">
> 
> step.2: open the [ Inspect:Panel ] drawer <img alt="image" width="1512" src="https://private-user-images.githubusercontent.com/30427789/314372035-9faf2ec2-4601-430d-aa3d-2b589f86af9c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTEzNTUwMTIsIm5iZiI6MTcxMTM1NDcxMiwicGF0aCI6Ii8zMDQyNzc4OS8zMTQzNzIwMzUtOWZhZjJlYzItNDYwMS00MzBkLWFhM2QtMmI1ODlmODZhZjljLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMjUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzI1VDA4MTgzMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTAxYjY3NWU0MjFkYTc3ZDZlMzRmNWZiNDgxNzczOTYzOGQzYjQzNGZlYjUzYmYzZDJlYWUwNjJlNGU1M2QxNmUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.Idqo1aCR_nRiO3BQRjVbsy8nxKrTHGG_67kVbVejfGc">
> 
> step.3: download CSV <img alt="image" width="1512" src="https://private-user-images.githubusercontent.com/30427789/314372210-4c56eb1c-ffa8-4376-b488-8d6d6959608a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTEzNTUwMTIsIm5iZiI6MTcxMTM1NDcxMiwicGF0aCI6Ii8zMDQyNzc4OS8zMTQzNzIyMTAtNGM1NmViMWMtZmZhOC00Mzc2LWI0ODgtOGQ2ZDY5NTk2MDhhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMjUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzI1VDA4MTgzMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTI2M2RmMDcxNTgzYTM1M2VhNmM5YWQ1Yjg5NWQxYmNjNmM5MjMwMWYxMWM4NmZmZjEzYTMzNTA3OTU0MzliNDAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.KzXHt0RSCq_1CqQmWwyw_CWgy3kIyZYVQDeEbqqlKfw">

Hi. Please see the attachment and I can provide more information if needed
[-data-2024-03-26 09_52_16.csv](https://github.com/deepflowio/deepflow/files/14751318/-data-2024-03-26.09_52_16.csv)

**Comment by twou12031** [comment]
> > THX for report.
> > Working on this issue, but need more information. Could you comment the data that has the same problem or send it to our community maintenance staff? This data will only be used for testing purposes.
> > Here's how you can find that data:
> > step.1: go to the view panel page <img alt="image" width="1509" src="https://private-user-images.githubusercontent.com/30427789/314371939-dcf338c0-083f-43a0-a205-99b4f69319c3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTEzNTUwMTIsIm5iZiI6MTcxMTM1NDcxMiwicGF0aCI6Ii8zMDQyNzc4OS8zMTQzNzE5MzktZGNmMzM4YzAtMDgzZi00M2EwLWEyMDUtOTliNGY2OTMxOWMzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMjUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzI1VDA4MTgzMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWJjODUzYzY5ZTE4MzQzOTcwZmU1ZDdmNTg1YzVjYjE2OTZiYzIzZjMwYWMzYjRjNGIyM2IwNTVhMDdiMjcwNGQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.6hcad-F7noP3CcFP3wwDRBs0wwSvIUxBmck3eLJ2YBE">
> > step.2: open the [ Inspect:Panel ] drawer <img alt="image" width="1512" src="https://private-user-images.githubusercontent.com/30427789/314372035-9faf2ec2-4601-430d-aa3d-2b589f86af9c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTEzNTUwMTIsIm5iZiI6MTcxMTM1NDcxMiwicGF0aCI6Ii8zMDQyNzc4OS8zMTQzNzIwMzUtOWZhZjJlYzItNDYwMS00MzBkLWFhM2QtMmI1ODlmODZhZjljLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMjUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzI1VDA4MTgzMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTAxYjY3NWU0MjFkYTc3ZDZlMzRmNWZiNDgxNzczOTYzOGQzYjQzNGZlYjUzYmYzZDJlYWUwNjJlNGU1M2QxNmUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.Idqo1aCR_nRiO3BQRjVbsy8nxKrTHGG_67kVbVejfGc">
> > step.3: download CSV <img alt="image" width="1512" src="https://private-user-images.githubusercontent.com/30427789/314372210-4c56eb1c-ffa8-4376-b488-8d6d6959608a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTEzNTUwMTIsIm5iZiI6MTcxMTM1NDcxMiwicGF0aCI6Ii8zMDQyNzc4OS8zMTQzNzIyMTAtNGM1NmViMWMtZmZhOC00Mzc2LWI0ODgtOGQ2ZDY5NTk2MDhhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMjUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzI1VDA4MTgzMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTI2M2RmMDcxNTgzYTM1M2VhNmM5YWQ1Yjg5NWQxYmNjNmM5MjMwMWYxMWM4NmZmZjEzYTMzNTA3OTU0MzliNDAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.KzXHt0RSCq_1CqQmWwyw_CWgy3kIyZYVQDeEbqqlKfw">
> 
> Hi. Please see the attachment and I can provide more information if needed [-data-2024-03-26 09_52_16.csv](https://github.com/deepflowio/deepflow/files/14751318/-data-2024-03-26.09_52_16.csv)

Got that.
Will fix it as soon as i can.

**Comment by twou12031** [comment]
Hi,
This bug has been fixed, please use the version 6.4.9 that we just re-released.

Why the bug happens:
Our deepflow server has some optimizations for find the related data.
Our deepflow-grafana-gui is not not adapted.

Our fix:
We have adapted it.

Thanks again for your report.
